downloadertask/IspwDownloader.ts

```diff
@@ -116,35 +121,38 @@ export class IspwDownloader {
     containers types(Assignment/Set/Release)
     */
     downloadContainerSource(cliArguments: any): void {
-        //Calling bat command
-        var bat = require.resolve(cliArguments.get("windowsTopazWorkbenchCliHome") + '/SCMDownloaderCLI.bat');
-        var ls = spawn(bat, ['-host', cliArguments.get("host"), '-port', cliArguments.get("port"), '-code', cliArguments.get("codePage"), '-id',
-            cliArguments.get("userId"), '-pass', cliArguments.get("password"), '-targetFolder', cliArguments.get("windowsSourceDownloadLocation"),
-            '-scm', 'ispwc', '-ispwContainerName', cliArguments.get("containerId"), '-ispwContainerType', cliArguments.get("container"),
-            '-ispwComponentType', cliArguments.get("componentType"), '-ispwDownloadAll', cliArguments.get("downloadUnchangedSource"),
-            '-ispwDownloadIncl', cliArguments.get("downloadIncludes"), '-ispwServerLevel', cliArguments.get("taskLevel"),
-            '-ispwServerConfig', cliArguments.get("runtimeConfig")
-        ]);
-        console.log(ls);
-
-        ls.stdout.on('data', function (data) {
-            console.log('stdout: ' + data);
-        });
-
-        ls.stderr.on('data', function (data) {
-            console.log('stderr: ' + data);
-        });
-
-        ls.on('exit', function (code) {
-            console.log('child process exited with code ' + code);
-        });
+        let command = this.getCommand(cliArguments);
+        console.log(command);
+        //Calling command
+        if (command != undefined) {
+            var ls = spawn(command, ['-host', cliArguments.get("host"), '-port', cliArguments.get("port"), '-code', cliArguments.get("codePage"), '-id',
+                cliArguments.get("userId"), '-pass', cliArguments.get("password"), '-targetFolder', cliArguments.get("sourceDownloadLocation"),
+                '-scm', 'ispwc', '-ispwContainerName', cliArguments.get("containerId"), '-ispwContainerType', cliArguments.get("container"),
+                '-ispwComponentType', cliArguments.get("componentType"), '-ispwDownloadAll', cliArguments.get("downloadUnchangedSource"),
+                '-ispwDownloadIncl', cliArguments.get("downloadIncludes"), '-ispwServerLevel', cliArguments.get("taskLevel"),
+                '-ispwServerConfig', cliArguments.get("runtimeConfig")
+            ]);
+            console.log(ls);
```

Comment:
is this console log required ?

Reply:
I think this will be helpful for the user to see the whole command and detect if there are any errors